fenix/app/src/main/java/org/mozilla/fenix/components/StoreProvider.kt

```diff
@@ -42,3 +45,37 @@ class StoreProviderFactory<T : Store<*, *>>(
         return StoreProvider(createStore()) as VM
     }
 }
+
+/**
+ * Helper function for creation of [StoreLazy].
+ */
+@MainThread
+fun <T : Store<*, *>> ViewModelStoreOwner.store(
```

Comment:
thoughts?
```suggestion
fun <T : Store<*, *>> ViewModelStoreOwner.lazyStore(
```

Reply:
Yeah makes sense, update to this and simplified the implementation.